Source/processcontainers/common/BaseAdministrator.h

```diff
@@ -20,59 +20,58 @@
 #pragma once
 
 #include "processcontainers/ProcessContainer.h"
+#include "BaseContainerIterator.h"
+#include "Lockable.h"
 
 namespace WPEFramework {
 namespace ProcessContainers {
 
-    template <typename TContainer, typename Mixin> // IContainerAdministrator Mixin
+    template <typename TContainer, typename Mixin> // IContainerAdministrator, Lockable Mixin
```

Comment:
Let's for now leave it like it is. I had a look and I understand what you are trying to achieve and we can have a chat about when the times comes again we can be in front of a whiteboard again. It is implementation behind an interface and we can always change it if we want. One remark perhaps, be a bit reluctant using mixins, they do add to the complexity of the software and not everybody might understand what you are trying to achieve